diff --git a/resources/biohackrxiv/latex.template b/resources/biohackrxiv/latex.template
index 7ddab70..4575ef8 100644
--- a/resources/biohackrxiv/latex.template
+++ b/resources/biohackrxiv/latex.template
@@ -33,9 +33,29 @@
     \endorigfigure
 }
 
-\newenvironment{CSLReferences}[2]%
-  {}%
-  {\par}
+\newlength{\cslhangindent}
+\setlength{\cslhangindent}{1.5em}
+\newlength{\csllabelwidth}
+\setlength{\csllabelwidth}{3em}
+\newlength{\cslentryspacingunit} % times entry-spacing
+\setlength{\cslentryspacingunit}{\parskip}
+\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
+ {% don't indent paragraphs
+  \setlength{\parindent}{0pt}
+  % turn on hanging indent if param 1 is 1
+  \ifodd #1
+  \let\oldpar\par
+  \def\par{\hangindent=\cslhangindent\oldpar}
+  \fi
+  % set entry spacing
+  \setlength{\parskip}{#2\cslentryspacingunit}
+ }%
+ {}
+\usepackage{calc}
+\newcommand{\CSLBlock}[1]{#1\hfill\break}
+\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
+\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}}
+\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
 
 \usepackage{fixltx2e} % provides \textsubscript
 \usepackage[
@@ -77,7 +97,7 @@
   \href@Orig{#1}{#2}\fi}
 \def\href#1#2{%
   \IfBeginWith{#1}{https://doi.org}%
-  {\href@Urllike{#1}{#2}}{\href@Notdoi{#1}{#2}}}
+  {\href@Orig{#1}{#2}}{\href@Notdoi{#1}{#2}}}
 \makeatother
 
 
